Add new unquarantine procedure + deprecate quarantine (#1985)
neo4j/docs-operations: modules/ROOT/pages/procedures.adoc

| *Mode* 3+| WRITE
|===

> [role=label--enterprise-edition label--admin-only label--deprecated-2025.01]

Reviewer: Not to forget: to add labels related to new Cypher versioning. Or add a note saying the following:
Cypher 5.26 (LTS) - the procedure is available and not deprecated
Cypher 5 in Neo4j 2025.01 - The procedure is deprecated
Cypher 25 in Neo4j 2025.01 - The procedure doesn’t exist

| *Mode* 3+| DBMS
|===

[NOTE]
====

> Users should prefer the usage of procedures.adoc#procedure_dbms_unquarantineDatabase[`dbms.unquarantineDatabase()`]

Reviewer: ```suggestion
Users should prefer the usage of <<procedure_dbms_unquarantineDatabase,`dbms.unquarantineDatabase()`>>
```

> over `dbms.quarantineDatabase()` because of the improvements and the new features (see the `operation` option ).

Reviewer: ```suggestion
```

> The deprecated `dbms.quarantineDatabase()` procedure is available in Cypher 5 but not in Cypher 25.

Reviewer: ```suggestion
The deprecated `dbms.quarantineDatabase()` procedure is available in Cypher 5, but not in Cypher 25.
```

====

[role=label--enterprise-edition label--admin-only]
[[procedure_dbms_unquarantineDatabase]]
=== dbms.unquarantineDatabase()

.Details
|===
| *Syntax* 3+m| dbms.unquarantineDatabase(server, databaseName, operation) :: ()
| *Description* 3+a| Remove the quarantine status from a database in a server performing a cleanup `operation`.
The possible values for the cleanup operation are:

* `keepStateKeepStore` -- do nothing; leave store and cluster state as they are.
* `replaceStateKeepStore` -- join as a new member, clearing the current cluster state but keeping the store.
* `replaceStateReplaceStore` -- join as a new member, clearing both the current cluster state and the store.
.4+| *Input arguments* | *Name* | *Type* | *Description*
| `server` | `STRING` | The name or the id of the server where there is the quarantined database.
| `databaseName` | `STRING` | The name of the database whose quarantine status must be removed.
| `operation` | `STRING` | Select a cleanup operation to perform while removing the quarantine.
| *Mode* 3+| DBMS
|===

[NOTE]
====
If you choose to clear the current cluster state, the server will try to join as a new member,

> but this joining can succeed if and only if there is a majority of old members "letting" the

Reviewer: ```suggestion
but this joining can succeed if and only if there is a majority of old members "letting" the new members in.
```

> new members in. Let's assume our cluster has a topology with three primaries. If there is only one server

Reviewer: ```suggestion
Let's assume our cluster has a topology with three primaries.
If there is only one server in `QUARANTINED` mode, then it is safe to choose `replaceStateKeepStore` or `replaceStateReplaceStore`.
```

> in `QUARANTINED` mode, then it is safe to choose `replaceStateKeepStore` or `replaceStateReplaceStore`.

Reviewer: ```suggestion
```

> If there are two servers in `QUARANTINED` mode, then you should not use concurrently `replaceStateKeepStore` or

Reviewer: ```suggestion
If there are two servers in `QUARANTINED` mode, then you should not use concurrently `replaceStateKeepStore` or `replaceStateReplaceStore` for both servers because there would be no majority to let them in.
```

> `replaceStateReplaceStore` for both servers because there would be no majority to let them in.

Reviewer: ```suggestion
```

Before Neo4j 5.23, the procedure can be run only with the `Admin` privileges. +
Replaced by xref:clustering/server-syntax.adoc#server-management-syntax[`ENABLE SERVER`].

// New in v6
| xref:procedures.adoc#procedure_dbms_quarantineDatabase[`dbms.quarantineDatabase()`]
| label:no[]
| label:yes[]
| label:admin-only[] label:deprecated[Deprecated in 2025.01] +

> Replaced by procedures.adoc#procedure_dbms_unquarantineDatabase[`dbms.unquarantineDatabase()`]

Reviewer: ```suggestion
Replaced by xref:procedures.adoc#procedure_dbms_unquarantineDatabase[`dbms.unquarantineDatabase()`]
```
